packages/pglite/src/base.ts

```diff
@@ -156,47 +156,75 @@ export abstract class BasePGlite
    */
   async #runQuery<T>(
     query: string,
-    params?: any[],
+    params: any[] = [],
     options?: QueryOptions,
   ): Promise<Results<T>> {
     return await this._runExclusiveQuery(async () => {
       // We need to parse, bind and execute a query with parameters
       this.#log('runQuery', query, params, options)
       await this._handleBlob(options?.blob)
-      const parsedParams =
-        params?.map((p) => serializeType(p, options?.setAllTypes)) || []
-      let results
+      
+      const results: BackendMessage[] = []
+
       try {
-        results = [
-          ...(await this.#execProtocolNoSync(
-            serialize.parse({
-              text: query,
-              types: parsedParams.map(([, type]) => type),
-            }),
-            options,
-          )),
-          ...(await this.#execProtocolNoSync(
-            serialize.bind({
-              values: parsedParams.map(([val]) => val),
-            }),
-            options,
-          )),
-          ...(await this.#execProtocolNoSync(
-            serialize.describe({ type: 'P' }),
+        for (const [msg] of await this.#execProtocolNoSync(
+          serialize.parse({
+            text: query,
+            types: params.map((p) => inferType(p)),
```

Comment:
A part of me wonders, why even bother trying to infer types in JS at all? Postgres knows better. We might as well remove this code that guesses types, and then have the serializers always expect an oid. Then we just serialize everything by oid.

```suggestion
```